provider/gateway/rest/client.go

```diff
@@ -293,6 +308,103 @@ func (c *client) LeaseStatus(ctx context.Context, id mtypes.LeaseID) (*cltypes.L
 	return &obj, nil
 }
 
+func (c *client) LeaseEvents(ctx context.Context, id mtypes.LeaseID, follow bool) (*LeaseKubeEvents, error) {
+	endpoint, err := url.Parse(c.host.String() + "/" + leaseEventsPath(id))
+	if err != nil {
+		return nil, err
+	}
+
+	switch endpoint.Scheme {
+	case "wss", "https":
+		endpoint.Scheme = "wss"
+	default:
+		return nil, errors.Errorf("invalid uri scheme \"%s\"", endpoint.Scheme)
+	}
+
+	query := url.Values{}
+	query.Set("follow", strconv.FormatBool(follow))
+
+	endpoint.RawQuery = query.Encode()
+	conn, response, err := c.wsclient.DialContext(ctx, endpoint.String(), nil)
+	if err != nil {
+		if errors.Is(err, websocket.ErrBadHandshake) {
+			buf := &bytes.Buffer{}
+			_, _ = io.Copy(buf, response.Body)
+
+			return nil, ClientResponseError{
+				Status:  response.StatusCode,
+				Message: buf.String(),
+			}
+		}
+
+		return nil, err
+	}
+
+	streamch := make(chan *eventsv1.Event)
```

Comment:
This should probably be buffered, if someone is trying to dump the output into a file it should run faster since Go doesn't have to context switch so often.